content/en/docs/instrumentation/java/automatic/spring-boot.md

```diff
@@ -134,6 +134,457 @@ dependencies {
+to learn more about aspect weaving in spring.
+
+#### Usage
+
+```java
+import org.springframework.stereotype.Component;
+
+import io.opentelemetry.instrumentation.annotations.SpanAttribute;
+import io.opentelemetry.instrumentation.annotations.WithSpan;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+
+/**
+ * Test WithSpan
+ */
+@Component
+public class TracedClass {
+
+    @WithSpan
+    public void tracedMethod() {
+    }
+
+    @WithSpan(value="span name")
+    public void tracedMethodWithName() {
+        Span currentSpan = Span.current();
+        currentSpan.addEvent("ADD EVENT TO tracedMethodWithName SPAN");
+        currentSpan.setAttribute("isTestAttribute", true);
+    }
+
+    @WithSpan(kind = SpanKind.CLIENT)
+    public void tracedClientSpan() {
+    }
+
+    public void tracedMethodWithAttribute(@SpanAttribute("attributeName") String parameter) {
+    }
+}
+
+```
+
+### Sample Traces
+
+The traces below were exported using Zipkin.
+
+#### Spring Web MVC - Server Span
+
+```json
+   {
+      "traceId":"0371febbbfa76b2e285a08b53a055d17",
+      "id":"9b782243ad7df179",
+      "kind":"SERVER",
+      "name":"webmvctracingfilter.dofilterinteral",
+      "timestamp":1596841405866633,
+      "duration":355648,
+      "localEndpoint":{
+         "serviceName":"sample_trace",
+         "ipv4":"XXX.XXX.X.XXX"
+      },
+      "tags":{
+         "http.client_ip":"0:0:0:0:0:0:0:1",
+         "http.flavor":"1.1",
+         "http.method":"GET",
+         "http.status_code":"200",
+         "http.url":"/spring-webmvc/sample",
+         "http.user_agent":"PostmanRuntime/7.26.2",
+         "net.sock.peer.addr":"0:0:0:0:0:0:0:1",
+         "net.sock.peer.port":"33916",
+         "net.sock.family":"inet6"
+         "sampling.probability":"1.0"
+      }
+   }
+```
+
+#### Spring Web - RestTemplate Client Span
+
+```json
+{
+  "traceId": "0371febbbfa76b2e285a08b53a055d17",
+  "parentId": "9b782243ad7df179",
+  "id": "43990118a8bdbdf5",
+  "kind": "CLIENT",
+  "name": "http get",
+  "timestamp": 1596841405949825,
+  "duration": 21288,
+  "localEndpoint": {
+    "serviceName": "sample_trace",
+    "ipv4": "XXX.XXX.X.XXX"
+  },
+  "tags": {
+    "http.method": "GET",
+    "http.status_code": "200",
+    "http.url": "/spring-web/sample/rest-template",
+    "net.peer.name": "localhost",
+    "net.peer.port": "8081"
+  }
+}
+```
+
+#### Spring Web-Flux - WebClient Span
+
+```json
+{
+  "traceId": "0371febbbfa76b2e285a08b53a055d17",
+  "parentId": "9b782243ad7df179",
+  "id": "1b14a2fc89d7a762",
+  "kind": "CLIENT",
+  "name": "http post",
+  "timestamp": 1596841406109125,
+  "duration": 25137,
+  "localEndpoint": {
+    "serviceName": "sample_trace",
+    "ipv4": "XXX.XXX.X.XXX"
+  },
+  "tags": {
+    "http.method": "POST",
+    "http.status_code": "200",
+    "http.url": "/spring-webflux/sample/web-client",
+    "net.peer.name": "localhost",
+    "net.peer.port": "8082"
+  }
+}
+```
+
+#### @WithSpan Instrumentation
```

Comment:
Everything related to `@WithSpan` could perhaps be moved to a specific section not related to the automatic instrumentation, perhaps in the _Additional instrumentations_ section?